client/src/pages/Suggest/Suggest.test.jsx

```diff
@@ -26,10 +26,16 @@ describe("Suggest", () => {
 		await user.type(screen.getByRole("textbox", { name: /title/i }), title);
 		await user.type(screen.getByRole("textbox", { name: /url/i }), url);
 		await user.click(screen.getByRole("button", { name: /suggest/i }));
+		if (requestBody.draft) {
+			await expect(
+				screen.findByText(/thank you for suggesting a resource/i)
+			).resolves.toHaveClass("message", "success");
+		} else {
+			await expect(
+				screen.findByText(/thank you for publishing a resource/i)
+			).resolves.toHaveClass("message", "success");
+		}
```

Comment:
❌ You should pretty much never have conditional logic in tests. Only _one_ of these cases is ever actually tested, there's no test where _"suggesting"_ would be shown (which is what made them start failing in the first place), so including the other branch just confuses the issue and implies that's tested when it isn't.

❌ It's also incorrect - whether it shows "suggesting" or "publishing" isn't a matter of the _request_ body (which never has a draft property - this should be clear from the assertions on it), it's a matter of the _response_ body (which is an empty object in all of these tests).

Reply:
ok I got that thanks